### ISSUE TITLE
Problem installing Yeoman - Failed at the yo@1.3.3 postinstall script

### ISSUE THREAD

**Issue by ferry77** [yeoman/yeoman#1439]
Hi,
I am trying to install yeoman, i use macbook pro with Yosemite OSX.
Below is the error.. 
On the other side, I can install grunt and bower without any problem.. 
After executing below command, there is no yo folder at all inside node_modules.
I tried 'npm uninstall -g yo && npm cache clean && npm i -g yo' as suggested on https://github.com/yeoman/yo/issues/128 but no luck yet.
Could you please help?

Thanks

```
$ npm install -g yo
/Users/theuser/.node/bin/yo -> /Users/theuser/.node/lib/node_modules/yo/cli.js

> yo@1.3.3 postinstall /Users/theuser/.node/lib/node_modules/yo
> yodoctor


module.js:340
    throw err;
          ^
Error: Cannot find module './doctor'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/Users/theuser/.node/lib/node_modules/yo/node_modules/yeoman-doctor/cli.js:4:1)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)

npm ERR! yo@1.3.3 postinstall: `yodoctor`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the yo@1.3.3 postinstall script.
npm ERR! This is most likely a problem with the yo package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     yodoctor
npm ERR! You can get their info via:
npm ERR!     npm owner ls yo
npm ERR! There is likely additional logging output above.
npm ERR! System Darwin 14.0.0
npm ERR! command "/Users/theuser/.node/bin/node" "/Users/theuser/.node/bin/npm" "install" "-g" "yo"
npm ERR! cwd /Users/theuser
npm ERR! node -v v0.10.33
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR! not ok code 0
```

**Comment by coderigo** [comment]
Having the same problem as @kylegoetz and others. FWIW @SBoudrias (or anyone else that can help troubleshoot), below is a Dockerfile to reproduce the error exactly. I have commented out `RUN npm install -g yo@1.3.3` as it trips up at that part, but firing up a container based on this image and at the terminal trying to fire the `npm install -g yo@1.3.3` reproduces the error. Similar failure with setting it to `@1.3.0`. 

Happy to provide any other info I can so we can help get past this.

```docker
# To build the image from this file, cd into the directory hosting this Dockerfile and type:
# docker build --tag="yobug-ui-img" ./

# Base image
FROM ubuntu:14.04

# Maintainer
MAINTAINER Rodrigo Martell <rodrigo.martell@gmail.com>

# Supress debian frontend warnings (weird)
RUN DEBIAN_FRONTEND=noninteractive

# Install system stuff
RUN apt-get update
RUN apt-get -y install curl
RUN apt-get -y install git

# Node stuff
# ref: https://github.com/hbouvier/docker-cloud9/blob/master/Dockerfile
RUN git clone git://github.com/creationix/nvm.git ~/.nvm
RUN /bin/bash -c '. ~/.nvm/nvm.sh && nvm install v0.10.34 && nvm use v0.10.34 && nvm alias default v0.10.34 && ln -s ~/.nvm/v0.10.34/bin/node /usr/bin/node && ln -s ~/.nvm/v0.10.34/bin/npm /usr/bin/npm'
RUN /bin/bash -c 'export PATH=./node_modules/.bin:$PATH'
RUN /bin/bash -c 'curl https://npmjs.org/install.sh -L -o ~/npm-install.sh && . ~/npm-install.sh && nvm use v0.10.34 && nvm alias default v0.10.34'

# Set npm to v2.1.18 and install yeoman
RUN npm install -g npm@2.1.18
# The next command fails with the error described in https://github.com/yeoman/yeoman/issues/1439
# so it is commented out, but can be reproduced once a container is run based on this image.
# RUN npm install -g yo@1.3.3

# Yeoman and livereload ports
EXPOSE 9000
EXPOSE 35729

# Set env variables
ENV HOME /

# Change working directory
WORKDIR /yobug-ui

# Default command when container started from image (start a shell)
CMD ["/bin/bash"]

# To fire up a container from this image, run this command:
# docker run --privileged=true --name="yobug-ui" -p 9000:9000 -p 35729:35729 -it -v $(pwd):/yobug-ui yobug-ui-img
```

**Comment by radiart** [comment]
Same problem on Windows 7:


npm ERR! yo@1.3.3 postinstall: `yodoctor`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the yo@1.3.3 postinstall script.
npm ERR! This is most likely a problem with the yo package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     yodoctor
npm ERR! You can get their info via:
npm ERR!     npm owner ls yo
npm ERR! There is likely additional logging output above.
npm ERR! System Windows_NT 6.1.7601
npm ERR! command "c:\\Program Files\\nodejs\\node.exe" "c:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "yo"
npm ERR! cwd D:\EasyPHP-DevServer-14.1VC11\data\localweb
npm ERR! node -v v0.10.36
npm ERR! npm -v 1.4.28
npm ERR! syscall spawn
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! not ok code 0

**Comment by mb2o** [comment]
Same issue on Windows 8.1 (x64)

**Comment by mb2o** [comment]
I'm really starting to get sick and tired of the nodejs/npm implementation for Windows

**Comment by SBoudrias** [comment]
By the way, you can all install `npm -g install yeoman-doctor` before installing yo if you can't figure out what broke npm.

**Comment by coderigo** [comment]
Thanks @SBoudrias, but no luck with that command issued before installing `yo`.
Here's the same Dockerfile with `yeoman-doctor` installed before `yo` for anyone's reference.:

```docker
# To build the image from this file, cd into the directory hosting this Dockerfile and type:
#     docker build --tag="yobug-ui-img" --rm=true ./
# To fire up a container from this image, run this command:
#     docker run --name="yobug-ui" -p 9000:9000 -p 35729:35729 -it -v $(pwd):/yobug-ui yobug-ui-img

# Base image
FROM ubuntu:14.04

# Supress debian frontend warnings (weird)
RUN DEBIAN_FRONTEND=noninteractive

# Install system stuff
RUN apt-get update
RUN apt-get -y install curl
RUN apt-get -y install git

# Node stuff
# ref: https://github.com/hbouvier/docker-cloud9/blob/master/Dockerfile
RUN git clone git://github.com/creationix/nvm.git ~/.nvm
RUN /bin/bash -c '. ~/.nvm/nvm.sh && nvm install v0.10.34 && nvm use v0.10.34 && nvm alias default v0.10.34 && ln -s ~/.nvm/v0.10.34/bin/node /usr/bin/node && ln -s ~/.nvm/v0.10.34/bin/npm /usr/bin/npm'
RUN /bin/bash -c 'export PATH=./node_modules/.bin:$PATH'
RUN /bin/bash -c 'curl https://npmjs.org/install.sh -L -o ~/npm-install.sh && . ~/npm-install.sh && nvm use v0.10.34 && nvm alias default v0.10.34'

# Set npm to v2.1.18 and install yeoman
RUN npm install -g npm@2.1.18

# Pre-install yeoman-doctor to stop `npm install -g yo@1.3.3` failing. 
RUN npm install -g yeoman-doctor

# The next command fails with the error described in https://github.com/yeoman/yeoman/issues/1439
RUN npm install -g yo@1.3.3

# Yeoman and livereload ports
EXPOSE 9000
EXPOSE 35729

# Set env variables
ENV HOME /

# Change working directory
RUN mkdir youbug-ui
WORKDIR /yobug-ui

# Default command when container started from image (start a shell)
CMD ["/bin/bash"]
```

**Comment by SBoudrias** [comment]
Well, only thing left is open an issue on npm.

**Comment by coderigo** [comment]
Thanks. Will do tonight unless someone gets there sooner.

**Comment by SBoudrias** [comment]
@coderigo also there's the usual check: make sure node is in your path, make sure npm binaries are in your path, check your NODE_PATH and npm prefix is the correct value. Your Dockerfile with `nvm` makes it unclear these are setup correctly.

On an unrelated notes, why do you run `yo` inside docker? I've actually never saw a development environment setup on a docker machine; I feel it is usually used for ops as part of easily deploying services. I wonder what is your use case and your goal with this setup.

**Comment by coderigo** [comment]
@SBoudrias . Thanks for that. I planned to make time last night but a bit snowed under.
Re running `yo` inside docker: I was doing this as part of a project that needed multiple people to make changes to a UI app and a RESTful backend in PHP. We wanted to make it easy for the project to have dev environments (one env  for front end, another for the backend) live with the code, so we had a go at making Dockerfiles with all the bits needed to develop defined in them. To develop the front end, for example, we could clone the repo, build the image based on the Dockerfile and fire up a container based on it with the code mounted into the container. I've done something like this previously and it works well but while setting up the Docker image this time I got stuck on this `yo-doctor` issue that brought me here. Anyway, a bit off-topic I guess, but happy to chat some more at the email listed in the first Dockerfile I posted (silly move, perhaps) in this thread.

Thanks again for your comments. I'll have another tinker and create issue at NPM if no luck. Hopefully will get to it this weekend as it seems a few of us are getting the same issue.

**Comment by radiart** [comment]
Ok I found solution!!! It's working! I use windows 7.

You must first update NPM to latest version to get YO working. 

The problem with NPM on Windows is when you use install pack, because you have NPM in two folders. I spent on this few hours but I finally updated NPM by doing these steps:

https://github.com/npm/npm/wiki/Troubleshooting#upgrading-on-windows

Make all options, and combine with WINDOWS PATH and command: "npm install npm" this is the key. Good luck
